src/main/java/com/yugabyte/sample/apps/CassandraEventData.java

```diff
@@ -0,0 +1,309 @@
+// Copyright (c) YugaByte, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied.  See the License for the specific language governing permissions and limitations
+// under the License.
+//
+
+package com.yugabyte.sample.apps;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.log4j.Logger;
+
+import com.datastax.driver.core.BatchStatement;
+import com.datastax.driver.core.BoundStatement;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+
+import com.yugabyte.sample.common.CmdLineOpts;
+import com.datastax.driver.core.utils.Bytes;
+
+/**
+ * A sample IoT event data application with batch processing.
+ *
+ * This app tracks a bunch of metrics which have a series of data points ordered
+ * by timestamp. The metric's id can be thought of as a compound/concatenated
+ * unique id - which could include user id, on a node id and device id (and id
+ * of the device). For simplicity, we use it as a single metric id to cover all
+ * of them.
+ */
+public class CassandraEventData extends AppBase {
+	private static final Logger LOG = Logger.getLogger(CassandraEventData.class);
+	// Static initialization of this workload's config.
+	static {
+		// Set the default number of reader and writer threads.
+		appConfig.numReaderThreads = 2;
+		appConfig.numWriterThreads = 16;
+		// Set the number of keys to read and write.
+		appConfig.numKeysToRead = -1;
+		appConfig.numKeysToWrite = -1;
+		// The size of the value.
+		appConfig.valueSize = 3000;
+		// Set the TTL for the raw table.
+		appConfig.tableTTLSeconds = 24 * 60 * 60;
+		// Default write batch size.
+		appConfig.cassandraBatchSize = 25;
+		// Default read batch size.
+		appConfig.cassandraReadBatchSize = 25;
+		// Default time delta from current time to read in a batch.
+		appConfig.readBackDeltaTimeFromNow = 180;
+	}
+
+	static Random random = new Random();
+	// The default table name that has the raw metric data.
+	private final String DEFAULT_TABLE_NAME = "batch_event_data_ts_metrics_raw";
+	// The structure to hold info per metric.
+	static List<DataSource> dataSources = new CopyOnWriteArrayList<DataSource>();
+	// The minimum number of metrics to simulate.
+	private static int min_metrics_count = 50;
+	// The maximum number of metrics to simulate.
+	private static int max_metrics_count = 100;
```

Comment:
You don't need a min and a max, just need num_devices.

Reply:
Done